crates/trigger-http/src/handler.rs

```diff
@@ -170,13 +171,37 @@ impl HttpHandlerExecutor {
             .data_mut()
             .new_response_outparam(response_tx)?;
 
-        let proxy = Proxy::new(&mut store, &instance)?;
+        enum Handler {
+            Latest(Proxy),
+            Handler2023_10_18(IncomingHandler2023_10_18),
+        }
+
+        let handler = match instance
+            .exports(&mut store)
+            .instance("wasi:http/incoming-handler@0.2.0-rc-2023-10-18")
+        {
+            Some(mut instance) => Some(Handler::Handler2023_10_18(IncomingHandler2023_10_18::new(
+                &mut instance,
+            )?)),
+            None => None,
+        };
+        let handler = match handler {
+            Some(handler) => handler,
+            None => Handler::Latest(Proxy::new(&mut store, &instance)?),
+        };
```

Comment:
```suggestion
        let handler = match instance
            .exports(&mut store)
            .instance("wasi:http/incoming-handler@0.2.0-rc-2023-10-18")
        {
            Some(mut instance) => Handler::Handler2023_10_18(IncomingHandler2023_10_18::new(
                &mut instance,
            )?),
            None => Handler::Latest(Proxy::new(&mut store, &instance)?),
        };
```

Reply:
Ah I tried this originally yeah but that runs afoul of `cannot borrow `store` as mutable more than once at a time` where the `instance.exports(...)` borrows the `store` for the whole `match` meaning that the `Proxy::new` call can't be part of the same `match`. Either that or I hand-wave and say "NLL isn't fully done yet and polonius will fix this" or something along those lines.